torch/csrc/jit/passes/onnx/constant_fold.cpp

```diff
@@ -265,6 +268,34 @@ c10::optional<at::Tensor> runTorchBackendForOnnx(
       }
     }
     return c10::optional<at::Tensor>(at::reshape(updated_val, shape));
+  } else if (node->kind() == onnx::ReduceL1 || node->kind() == onnx::ReduceL2) {
+    assert(inputTensorValues.size() == 1);
+    if (!node->hasAttributeS("axes")) {
+      return c10::nullopt;
+    }
+    if (!node->hasAttributeS("keepdims")) {
+      return c10::nullopt;
+    }
+    int p = node->kind() == onnx::ReduceL1 ? 1 : 2;
+    updated_val = at::norm(inputTensorValues[0], p, node->is(attr::axes), node->i(attr::keepdims));
+    return c10::optional<at::Tensor>(updated_val);
+  } else if (node->kind() == onnx::Gather) {
+    assert(inputTensorValues.size() == 1);
+    if (!node->hasAttributeS("axis")) {
+      return c10::nullopt;
+    }
+    auto axis = node->i(attr::axis);
+    // If axis attribute for onnx::Gather has a value less than 0,
+    // It needs to be adjusted (+= dim sizes) for aten op
+    axis += axis < 0 ? inputTensorValues[0].sizes().size() : 0;
+    at::Tensor indices = inputTensorValues[1];
+    // If indices input for onnx::Gather has a value less than 0,
+    // It needs to be adjusted (+= dim value) for aten op
+    auto less_mask = at::lt(indices, 0);
+    auto indices_corr = at::add(indices, inputTensorValues[0].sizes()[axis]);
+    auto indices_masked = at::where(less_mask, indices_corr, indices);
```

Comment:
I am not sure to understand what is happening here.
My understanding is that :
- indices contains positive and negative indices
- less mask is a mask to identify the negative indices
- indices_corr is the "corrected" indices; inputTensorValues[0].sizes()[axis] is added for all indices even if they are positive ?
- indices_masked is a mask on the corrected negative indices ?

Reply:
The "where" op, takes values from "indices_corr" where condition (indices < 0) is true, and takes values from "indices" where condition is false. So in cases where indices < 0, we have indices += dim value.

Reply:
do we have a test covering this pass?

Reply:
Yes, this is in test_utility_functions line 157. Index is < 0